packages/enzyme-test-suite/test/ReactWrapper-spec.jsx

```diff
@@ -1706,6 +1706,33 @@ describeWithDOM('mount', () => {
       expect(foundNotSpan).to.have.lengthOf(0);
     });
 
+    it('does not get trapped when conditionally rendering using an empty string variable as the condition', () => {
+      const emptyString = '';
+
+      class Foo extends React.Component {
+        render() {
+          return (
+            <div>
+              <header>
+                <span />
+                {emptyString && <i />}
+              </header>
+              <div>
+                <span data-foo={this.props.selector}>Test</span>
+              </div>
+            </div>
+          );
+        }
+      }
+
+      const selector = 'blah';
+      const wrapper = mount(<Foo selector={selector} />);
+      const foundSpan = wrapper.findWhere(n => (
+        n.type() === 'span' && n.props()['data-foo'] === selector
+      ));
+      expect(foundSpan.type()).to.equal('span');
```

Comment:
it might be more robust here to instead make an assertion on `foundSpan.debug()`?